jdbc-core/src/main/java/com/salesforce/datacloud/jdbc/core/DataCloudStatement.java

```diff
@@ -150,17 +153,30 @@ public int getMaxFieldSize() {
     @Override
     public void setMaxFieldSize(int max) {}
 
-    @Getter(AccessLevel.PACKAGE)
-    private int targetMaxRows;
-
-    @Getter(AccessLevel.PACKAGE)
-    private int targetMaxBytes;
+    // The maximum byte size limit for a row based RPC response. While the server enforces a max as well, also having
+    // the constant available on the client side allows to set appropriate default values and also to provide immediate
+    // feedback on the
+    // ``setResultSetConstraints`` method.
+    public static final int HYPER_MAX_ROW_LIMIT_BYTE_SIZE = 20971520;
+    // The minimal byte size limit for a row based RPC response. The driver enforces this to guard against code that
+    // accidentally provides the limit in megabytes.
+    public static final int HYPER_MIN_ROW_LIMIT_BYTE_SIZE = 1024;
 
     public void clearResultSetConstraints() throws DataCloudJDBCException {
         setResultSetConstraints(0, HYPER_MAX_ROW_LIMIT_BYTE_SIZE);
     }
 
-    private void checkResultSetConstraints(int maxRows, int maxBytes) throws DataCloudJDBCException {
+    /**
+     * Sets the constraints that would limit the number of rows and overall bytes in any ResultSet object generated by this Statement.
+     * This is used to tell the database the maximum number of rows and bytes to return, you may get less than expected because of this.
+     *
+     * @param maxRows The target maximum number of rows a ResultSet can have, zero means there is no limit.
```

Comment:
Afaict, we will never return more than those?

```suggestion
     * @param maxRows The maximum number of rows a ResultSet can have, zero means there is no limit.
```